code/game/machinery/firealarm.dm

```diff
@@ -326,3 +326,185 @@
 	if (!party_overlay)
 		party_overlay = iconstate2appearance('icons/turf/areas.dmi', "party")
 	A.add_overlay(party_overlay)
+
+/obj/machinery/sprinkler
+	name = "sprinkler"
+	desc = "An automated sprinkler capable of detecting fire and spraying coolant."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "sprinkler0"
+	layer = PROJECTILE_HIT_THRESHHOLD_LAYER
+	plane = FLOOR_PLANE
+	max_integrity = 100
+	armor = list("melee" = 50, "bullet" = 20, "laser" = 20, "energy" = 20, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 80, "acid" = 30)
+	req_access = list(ACCESS_ENGINE)
+	var/refill_loaded = FALSE
+	var/cooldown = 0
+	var/working = FALSE
+
+/obj/machinery/sprinkler/Initialize()
+	. = ..()
+	create_reagents(50)
+
+/obj/machinery/sprinkler/interact(mob/user)
+	if(!allowed(user))
+		to_chat(user, "<span class='danger'>Access denied.</span>")
+		return
+	dispense()
+	log_game("[user] has manually actived the sprinkler in [AREACOORD(src)]")
+
+/obj/machinery/sprinkler/AltClick(mob/user)
+	if(issilicon(user))
+		var/harm = alert(user, "Unsanitized input detected, do you want to use this exploit to sabotage the sprinkler chemical storage? It will lead to human harm.", "root", "Abort", "Proceed")
+		if(harm != "Proceed")
+			return
+		emag_act(user)

```

Comment:
If these were common I would be worried about this being really powerful for malf AIs or syndiborgs, why should borgs get a free emag on this one item specifically?